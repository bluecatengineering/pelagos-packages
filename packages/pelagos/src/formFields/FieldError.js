import PropTypes from 'prop-types';

import './FieldError.less';

/** @deprecated use FieldHelper with an error message. */
const FieldError = ({text, ...props}) => (
	<div {...props} className="FieldError" aria-live="polite">
		{text || '\u00a0'}
	</div>
);

FieldError.propTypes = {
	/** The component id. */
	id: PropTypes.string,
	/** The error text. */
	text: PropTypes.string,
};

export default FieldError;
