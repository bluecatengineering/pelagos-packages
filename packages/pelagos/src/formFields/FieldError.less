@import '../../less/fonts';
@import '../../less/spacing';

@layer pelagos {
	.FieldError {
		@font-12-400();
		color: var(--text-error);
		margin-top: @sp-04;
	}
}
