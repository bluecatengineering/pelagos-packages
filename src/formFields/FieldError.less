@import '../../less/colors';
@import '../../less/fonts';

.FieldError {
	@font-10-600-uc();
	height: 1.5em;
}
