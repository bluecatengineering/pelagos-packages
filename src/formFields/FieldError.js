import React from 'react';
import PropTypes from 'prop-types';

import './FieldError.less';

/** An error message. */
const FieldError = ({text, alignment, ...props}) => (
	<div {...props} className="FieldError" style={{textAlign: alignment}}>
		{text}
	</div>
);

FieldError.propTypes = {
	/** The component id. */
	id: PropTypes.string,
	/** The error text. */
	text: PropTypes.string,
	/** The text alignment. */
	alignment: PropTypes.oneOf(['left', 'right']),
};

FieldError.defaultProps = {
	alignment: 'right',
};

export default FieldError;
