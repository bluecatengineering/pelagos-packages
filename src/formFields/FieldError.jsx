import React from 'react';
import PropTypes from 'prop-types';

import './FieldError.less';

const FieldError = ({componentId, text, alignment}) => (
	<div className="FieldError" data-bcn-id={componentId || 'input-error'} style={{textAlign: alignment}}>
		{text}
	</div>
);

FieldError.propTypes = {
	componentId: PropTypes.string,
	text: PropTypes.string,
	alignment: PropTypes.oneOf(['left', 'right']),
};

FieldError.defaultProps = {
	alignment: 'right',
};

export default FieldError;
