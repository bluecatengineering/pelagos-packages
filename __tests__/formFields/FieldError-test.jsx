import React from 'react';
import {shallow} from 'enzyme';

import FieldError from '../../src/formFields/FieldError';

jest.unmock('../../src/formFields/FieldError');

describe('FieldError', () => {
	describe('rendering', () => {
		it('renders expected elements', () => {
			const wrapper = shallow(<FieldError text="Test" />);
			expect(wrapper.getElement()).toMatchSnapshot();
		});

		it('renders expected elements when componentId is set', () => {
			const wrapper = shallow(<FieldError componentId="test" text="Test" />);
			expect(wrapper.getElement()).toMatchSnapshot();
		});
	});
});
