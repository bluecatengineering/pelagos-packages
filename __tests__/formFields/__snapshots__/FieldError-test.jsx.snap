// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldError rendering renders expected elements 1`] = `
<div
  className="FieldError"
  data-bcn-id="input-error"
  style={
    Object {
      "textAlign": "right",
    }
  }
>
  Test
</div>
`;

exports[`FieldError rendering renders expected elements when componentId is set 1`] = `
<div
  className="FieldError"
  data-bcn-id="test"
  style={
    Object {
      "textAlign": "right",
    }
  }
>
  Test
</div>
`;
